contrib/palantirjavaformat/src/mill/contrib/palantirjavaformat/PalantirJavaFormatModule.scala

```diff
@@ -0,0 +1,187 @@
+package mill
+package contrib.palantirjavaformat
+
+import mill.api.Loose
+import mill.api.PathRef
+import mill.define.{Discover, ExternalModule}
+import mill.main.Tasks
+import mill.scalalib.DepSyntax
+import mill.scalalib.JavaModule
+import mill.util.Jvm
+
+/**
+ * Formats Java source files using [[https://github.com/palantir/palantir-java-format Palantir Java Format]].
+ */
+trait PalantirJavaFormatModule extends JavaModule {
+
+  /**
+   * Formats Java source files.
+   *
+   * @param check if an exception should be raised when formatting errors are found
+   *              - when `true`, files are not formatted
+   * @param sources list of file or folder path(s) to be processed
+   *                - path must be relative to [[millSourcePath]]
+   *                - when empty, all [[sources]] are processed
+   */
+  def javafmt(
```

Comment:
For consistency with the rest of Mill plugin traits, we should name this `def palantirformat`, so it matches the name of the trait that's providing it (unless it's a "shared" task that many different traits would use)